### PR TITLE
feat(466411): Use case sensitivity flag with new file uploads

### PR DESCRIPTION
**Pull request by mokhld**
This PR introduces case sensitivity handling for file retrieval keys (notification emails) throughout the Forms Submission API, from file ingestion, persistence and retrieval.

1. Case Sensitivity Detection
   - Added new helper function to determine if retrieval keys contain uppercase characters
   - Stores this information alongside the hashed key in MongoDB records

2. Database Schema
   - Added `retrievalKeyIsCaseSensitive` flag to track whether keys should be matched exactly or case-insensitively
   - Applied to all file operations: ingestion, submission, and persistence

3. Documentation updates and tests.

The goal of these changes is to grant us flexible file retrieval by maintaining backward compatibility with existing mixed-case keys while promoting consistent lowercase usage for new submissions. The `retrievalKeyIsCaseSensitive` flag facilitates the consuming forms-designer services to handle case matching appropriately when retrieval files.

### CODE REVIEW

**Comment by colinrotherham** [comment]
Hey @mokhld, if you can give this PR a rebase with `main`

The status checks have been renamed in [check-pull-request.yml](https://github.com/DEFRA/forms-submission-api/actions/workflows/check-pull-request.yml)

e.g. Status check  ~`ESLint (ubuntu-latest)`~ `ESLint`

**Review by alexluckett — COMMENTED** [review]
LGTM but will have a run of this locally on Monday

File: src/api/files/service.js [hunk 1]
```diff
@@ -397,6 +410,10 @@ async function getAndVerify(fileId, retrievalKey) {
     throw Boom.notFound('File not found')
   }
 
+  logger.info(
+    `Comparing retrieval keys - incoming: ${retrievalKey}, stored: ${fileStatus.retrievalKey}`
+  )
```
Comment (alexluckett) [inline src/api/files/service.js:415]: Don't log this, treat it as a credential

**Review by alexluckett — CHANGES_REQUESTED** [review]

File: src/api/files/service.js (hunk 1 above)
Comment (alexluckett) [reply src/api/files/service.js:415]: Other than that it's good to go

Comment (mokhld) [reply src/api/files/service.js]: 😱  Oversight on my part- my punishment for not setting up the debugger properly

**Review by alexluckett — APPROVED** [review]

**Pull request merged**